packages/telegram-cloud-storage-stamper/src/index.ts

```diff
@@ -0,0 +1,224 @@
+/// <reference lib="dom" />
+import { ApiKeyStamper } from "@turnkey/api-key-stamper";
+import { TelegramCloudStorageStamperError } from "./errors";
+
+declare global {
+  interface Window {
+    Telegram: any;
+  }
+}
+
+// config when telegram stamper is passed in an api key
+export type TTelegramCloudStorageStamperConfig = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+  cloudStorageKey?: string;
+};
+
+type CloudStorageAPIKey = {
+  apiPublicKey: string;
+  apiPrivateKey: string;
+};
+
+// Constant for default key name
+const DEFAULT_TURNKEY_CLOUD_STORAGE_KEY = "TURNKEY_API_KEY";
+
+/**
+ * Stamper to use within a `TurnkeyClient`
+ */
+export default class TelegramCloudStorageStamper {
+  // This stamper uses a typical apikey stamper under the hood and abstracts away the storage of the actual api keys
+  stamper: ApiKeyStamper | undefined;
+  cloudStorageKey: string | undefined;
+
+  private constructor(config: TTelegramCloudStorageStamperConfig) {
+    // check to see if were in a telegram mini app context
+    TelegramCloudStorageStamper.checkTelegramContext();
+
+    // instantiate the stamper
+    this.stamper = new ApiKeyStamper({
+      apiPublicKey: config.apiPublicKey,
+      apiPrivateKey: config.apiPrivateKey,
+    });
+
+    // set the cloud storage key
+    this.cloudStorageKey = config.cloudStorageKey;
+  }
+
+  // create a telegram stamper by getting/setting the private/public api key values from/to telegram cloud storage
+  static async create(
+    config?: TTelegramCloudStorageStamperConfig
+  ): Promise<TelegramCloudStorageStamper> {
+    // check to see if were in a telegram mini app context
+    TelegramCloudStorageStamper.checkTelegramContext();
+
+    let cloudStorageKey = DEFAULT_TURNKEY_CLOUD_STORAGE_KEY;
+    let apiPublicKey = "";
+    let apiPrivateKey = "";
+
+    // set the api pub and priv key if config is passed in
+    if (config) {
+      apiPublicKey = config.apiPublicKey;
+      apiPrivateKey = config.apiPrivateKey;
+      if (config.cloudStorageKey) {
+        // set the cloud storage key if it is passed in
+        cloudStorageKey = config.cloudStorageKey;
+      }
+
+      try {
+        await TelegramCloudStorageStamper.setCloudStorageItem(
+          cloudStorageKey,
+          TelegramCloudStorageStamper.stringifyAPIKey(
+            config.apiPublicKey,
+            config.apiPrivateKey
+          )
+        );
+      } catch (e) {
```

Comment:
worth templating `e.toString()` in the error message maybe? There could be multiple types of failures underneath, might be useful to know what happened exactly.

Reply:
The reason I didnt display the error message here specifically is that the getter and setter error messages that are thrown display the key and value that were failed to be input. This would display the pub/priv api key which i think in this case is a no no. However maybe I just shouldnt have the key/value printed in the underlying error message, what do you think?